_posts/2020-09-18-docker-params-change-detection.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "Docker params change detection"
+author: adrianiurca
+categories:
+  - docker
+  - development
+tags:
+  - bugfix
+  - docker
+  - personal
+---
+
+## Docker params change detection
+
+An interesting behaviour was present in docker::run component. The problem was that if any parameter was added/modified/removed puppet agent didn't apply the change only if you stopped, removed manually the container, and reapply the manifest, forcing a new container creation. 
+
+The solution was to create a new function that detects if at least one parameter is changed. The detection mechanism is based on the check between parameter values from the manifest file and correspondent field from the docker inspect object of the currently running container. The solution is present in the 3.11.0 version. 
```

Comment:
```suggestion
The solution was to create a new function that detects if at least one parameter is changed. The detection mechanism is based on the check between parameter values from the manifest file and correspondent field from the docker inspect object of the currently running container. The solution is present in the `3.11.0` version. 
```

Reply:
I commited your changes. Thank you @sanfrancrisko